tests/test_webhook.py

```diff
@@ -0,0 +1,224 @@
+            "batch": True,
+            "parameters": {
+                "foo": "resources[0].name"
+            }
+        }
+
+        wh = Webhook(data=data, manager=self.get_manager())
+        wh.process(resources)
+        req = request_mock.call_args[1]
+
+        self.assertEqual("http://foo.com?foo=test_name", req['url'])
+        self.assertEqual("GET", req['method'])
+        self.assertEqual({}, req['headers'])
+
+    @mock.patch('c7n.actions.webhook.requests.Request')
+    @mock.patch('c7n.actions.webhook.requests.Session')
+    def test_process_batch_body(self, session_mock, request_mock):
+        resources = [
+            {
+                "name": "test_name",
+                "value": "test_value"
+            }
+        ]
+
+        data = {
+            "url": "http://foo.com",
+            "batch": True,
+            "body": "resources[].name",
+            "body-size": 10,
+            "headers": {
+                "test": "`header`"
+            },
+            "parameters": {
+                "foo": "resources[0].name"
+            }
+        }
+
+        wh = Webhook(data=data, manager=self.get_manager())
+        wh.process(resources)
+        req = request_mock.call_args[1]
+
+        self.assertEqual("http://foo.com?foo=test_name", req['url'])
+        self.assertEqual("POST", req['method'])
+        self.assertEqual(b'["test_name"]', req['data'])
+        self.assertEqual(
+            {"test": "header", "Content-Type": "application/json"},
+            req['headers'])
+
+    @mock.patch('c7n.actions.webhook.requests.Request')
+    @mock.patch('c7n.actions.webhook.requests.Session')
+    def test_process_no_batch(self, session_mock, request_mock):
+        resources = [
+            {
+                "name": "test1",
+                "value": "test_value"
+            },
+            {
+                "name": "test2",
+                "value": "test_value"
+            }
+        ]
+
+        data = {
+            "url": "http://foo.com",
+            "batch": False,
+            "parameters": {
+                "foo": "resource.name"
+            }
+        }
+
+        wh = Webhook(data=data, manager=self.get_manager())
+        wh.process(resources)
+        req1 = request_mock.call_args_list[0][1]
+        req2 = request_mock.call_args_list[1][1]
+
+        self.assertEqual("http://foo.com?foo=test1", req1['url'])
+        self.assertEqual("http://foo.com?foo=test2", req2['url'])
+
+    @mock.patch('c7n.actions.webhook.requests.Request')
+    @mock.patch('c7n.actions.webhook.requests.Session')
+    def test_process_existing_query_string(self, session_mock, request_mock):
+        resources = [
+            {
+                "name": "test1",
+                "value": "test_value"
+            },
+            {
+                "name": "test2",
+                "value": "test_value"
+            }
+        ]
+
+        data = {
+            "url": "http://foo.com?existing=test",
+            "batch": False,
```

Comment:
(minor) since `batch` is optional, I would just remove it from the `data` dictionary. (Same as below)